tests/jax/components/building/parameter_client_test.py

```diff
@@ -0,0 +1,315 @@
+    """Create a mock builder for testing that has a parameter client."""
+
+    builder = Builder(components=[])
+
+    builder.store.networks = {
+        "networks": {
+            "network_agent_0": SimpleNamespace(params={"weights": 0, "biases": 0}),
+            "network_agent_1": SimpleNamespace(params={"weights": 1, "biases": 1}),
+            "network_agent_2": SimpleNamespace(params={"weights": 2, "biases": 2}),
+        }
+    }
+
+    builder.store.trainer_networks = {
+        "trainer_0": ["network_agent_0", "network_agent_1", "network_agent_2"]
+    }
+    builder.store.trainer_id = "trainer_0"
+
+    builder.store.parameter_server_client = ParameterServer(
+        store=SimpleNamespace(get_parameters={"key_1": 1}),
+        components=[],
+    )
+
+    return builder
+
+
+def test_base_parameter_client() -> None:
+    """Test that count parameters are create in base \
+        parameter client"""
+
+    mock_client = MockBaseParameterClient(config=SimpleNamespace())
+
+    keys, params = mock_client._set_up_count_parameters(params={})
+
+    assert keys == [
+        "trainer_steps",
+        "trainer_walltime",
+        "evaluator_steps",
+        "evaluator_episodes",
+        "executor_episodes",
+        "executor_steps",
+    ]
+
+    assert params == {
+        "trainer_steps": np.array(0, dtype=np.int32),
+        "trainer_walltime": np.array(0, dtype=np.float32),
+        "evaluator_steps": np.array(0, dtype=np.int32),
+        "evaluator_episodes": np.array(0, dtype=np.int32),
+        "executor_episodes": np.array(0, dtype=np.int32),
+        "executor_steps": np.array(0, dtype=np.int32),
+    }
+
+
+def test_executor_parameter_client_no_evaluator_with_parameter_client(
+    mock_builder_with_parameter_client: Builder,
+) -> None:
+    """Test executor parameter client.
+
+    Args:
+        mock_builder_with_parameter_client : mava builder object
+    """
+
+    mock_builder = mock_builder_with_parameter_client
+    mock_builder.store.is_evaluator = False
+    exec_param_client = ExecutorParameterClient(
+        config=ExecutorParameterClientConfig(executor_parameter_update_period=100)
+    )
+    exec_param_client.on_building_executor_parameter_client(builder=mock_builder)
+
+    assert mock_builder.store.executor_parameter_client._all_keys == [
+        "evaluator_episodes",
+        "evaluator_steps",
+        "executor_episodes",
+        "executor_steps",
+        "networks-network_agent_0",
+        "networks-network_agent_1",
+        "networks-network_agent_2",
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_builder.store.executor_parameter_client._get_keys == [
+        "networks-network_agent_0",
+        "networks-network_agent_1",
+        "networks-network_agent_2",
+        "trainer_steps",
+        "trainer_walltime",
+        "evaluator_steps",
+        "evaluator_episodes",
+        "executor_episodes",
+        "executor_steps",
+    ]
```

Comment:
I think it might be a bit better to do:

```
expected_keys = set([
        "evaluator_episodes",
        "evaluator_steps",
        "executor_episodes",
        "executor_steps",
        "networks-network_agent_0",
        "networks-network_agent_1",
        "networks-network_agent_2",
        "trainer_steps",
        "trainer_walltime",
    ]
)

assert all([key in expected_keys for key in mock_builder.store.executor_parameter_client._all_keys])
assert all([key in expected_keys for key in mock_builder.store.executor_parameter_client._get_keys])
```

Unless I'm missing something but it seems like `get_keys` and `all_keys` are the same, but in a different order. So this should do the same, but it's a bit shorter and order agnostic.

Reply:
Totally agree. This definitely cleans things up really well. I've done this for all test cases.